contracts/koh/contracts/KingAutomaton.sol

```diff
@@ -369,6 +381,34 @@ contract KingAutomaton {
     }
   }
 
+  function claimReward(uint256 _id, uint256 _budget) public validBallotBoxID(_id) {
```

Comment:
Might be good to re-evaluate the proposal state at the start of this function, in case some time limit passed or something else changed.
If state is not re-evaluated, contributors may not be able to claim when they should, or abuse the system and claim when they shouldn't be able to.